### PR TITLE
Kdbai v1.4

### PR DESCRIPTION
**Pull request by alexgiannak**
Update KDBAI to version 1.4 with breaking changes

This PR updates the KDBAI integration to version 1.4, which includes significant changes to the kdbai-client API. These updates are necessary to maintain compatibility with the upcoming cloud and server versions, set to be released on Monday, October 21st.

Key changes:
- Updated kdbai-client API calls to align with the new 1.4 specification
- Adjusted data structures and response handling to accommodate API changes
- Updated error handling and logging to reflect  the changes

Important notes:
1. This update contains breaking changes. Projects using the current KDBAI integration will need to be updated.
2. Tests have been updated and I also have performed a self-review of my own code

### CODE REVIEW

File: requirements/connectors/kdbai.txt
Comment (rbiseck3) [inline requirements/connectors/kdbai.txt:9]: If this code is specific to this version, let's also add a `kdbai-client>=1..4.0` to the associated `.in` file.

**Comment by rbiseck3** [comment]
Will need to update the changelog and bump the version to pass the CI check, as this is required of every PR.

**Comment by rbiseck3** [comment]
I've added the `needs edits` tag on this PR for now. Feel free to remove that when this is ready to be rereviewed.

**Comment by rbiseck3** [comment]
I also realized that `kdbai-client==1.4.0` doesn't exist yet, we'll need to wait until that's published to merge this in.

**Comment by alexgiannak** [comment]
Hi Roman, 
I bumped the changelog.md and explicitly set the kdbai-client dep in the kdbai.in file as instructed. I think we are good to go. The version of the new server - client pair will be released on Monday

**Review by rbiseck3 — APPROVED** [review]

**Comment by rbiseck3** [comment]
Closed out in favor of this: https://github.com/Unstructured-IO/unstructured-ingest/pull/187

**Pull request closed without merging**
